XSConsoleData.py

```diff
@@ -528,17 +552,101 @@ def SaveToNTPConf(self):
         # Double-check authentication
         Auth.Inst().AssertAuthenticated()
 
-        file = None
         try:
-            file = open("/etc/chrony.conf", "w")
-            for other in self.ntp.othercontents([]):
-                file.write(other+"\n")
-            for server in self.ntp.servers([]):
-                file.write("server "+server+" iburst\n")
+            with open("/etc/chrony.conf", "w") as confFile:
+                for other in self.ntp.othercontents([]):
+                    confFile.write(other + "\n")
+                for server in self.ntp.servers([]):
+                    confFile.write("server " + server + " iburst\n")
         finally:
-            if file is not None: file.close()
             self.UpdateFromNTPConf()
 
+        # Force chronyd to update the time
+        if self.data['ntp']['method'] != "Disabled":
+            servers = self.data['ntp']['servers']
+            if self.data['ntp']['method'] == "DHCP":
+                servers = self.GetDHClientInterfaces()
+
+            if servers:
+                self.StopService("chronyd")
```

Comment:
I think it would be preferable to use `chronyc` to talk to the daemon and call e.g. the `makestep` command rather than stopping the daemon, running it in one shot mode, then restarting the daemon.

Reply:
Thanks! I didn't see that option in the man page the first time round! Updated!